test/sslapitest.c

```diff
@@ -625,10 +625,14 @@ static int test_client_cert_verify_cb(void)
 end:
     X509_free(crt1);
     X509_free(crt2);
-    SSL_shutdown(clientssl);
-    SSL_shutdown(serverssl);
-    SSL_free(serverssl);
-    SSL_free(clientssl);
+    if (clientssl != NULL) {
+        SSL_shutdown(clientssl);
```

Comment:
Why these dont do a NULL check internally I do not know

Reply:
It's not a _free() call, so passing NULL would be a naughty programmer error.

Reply:
Which means you end up doing ifs in your app instead :)